docs/2-building-blocks-of-oop-part-1/hometask/src/Triangle.ts

```diff
@@ -1,3 +1,24 @@
+import { Point } from './Point';
 import { Shape } from './Shape';
 
-export class Triangle extends Shape {}
+export class Triangle extends Shape {
+
+  public constructor(points: Point[]);
+  public constructor(points: Point[], color: string, filled: boolean);
+  public constructor(...args: any[]) {
+    super(args);
+  }
+
+  public toString(): string {
+    return `Triangle[v1=${this.points[0]},v2=${this.points[1]},v3=${this.points[2]}]`
+  }
+
+  public getType(): string {
+    const sideOne = this.points[0].distance(this.points[1]);
+    const sideTwo = this.points[1].distance(this.points[2]);
+    const sideTree = this.points[2].distance(this.points[0]);
+    if (sideOne === sideTwo && sideOne === sideTree) return "equilateral triangle"
```

Comment:
in the end there sin't semicolon.

Reply:
it's not necessary but if you want I can add it
https://www.google.com/search?q=do+we+need+to+use+semicolons+in+typescript&newwindow=1&rlz=1C1GTPM_enRU969RU970&ei=2PnLYoC7JYO8kwWIn7vIBA&oq=do+we+need+to+use+semicolons+in+ty&gs_lcp=Cgdnd3Mtd2l6EAMYADIFCCEQoAE6BwgAEEcQsAM6BwgAELADEEM6BQgAEIAEOgsILhCABBDHARCjAjoFCC4QgAQ6CwguEIAEEMcBENEDOggILhCABBDUAjoECAAQQzoLCC4QgAQQxwEQrwE6BggAEB4QFjoICCEQHhAWEB06CgghEB4QDxAWEB06BwghEAoQoAFKBAhBGABKBAhGGABQ2gpYxmFglm1oBHABeACAAa4BiAHoHZIBBTE1LjIwmAEAoAEByAEJwAEB&sclient=gws-wiz

Reply:
add pls

Reply:
> it's not necessary but if you want I can add it https://www.google.com/search?q=do+we+need+to+use+semicolons+in+typescript&newwindow=1&rlz=1C1GTPM_enRU969RU970&ei=2PnLYoC7JYO8kwWIn7vIBA&oq=do+we+need+to+use+semicolons+in+ty&gs_lcp=Cgdnd3Mtd2l6EAMYADIFCCEQoAE6BwgAEEcQsAM6BwgAELADEEM6BQgAEIAEOgsILhCABBDHARCjAjoFCC4QgAQ6CwguEIAEEMcBENEDOggILhCABBDUAjoECAAQQzoLCC4QgAQQxwEQrwE6BggAEB4QFjoICCEQHhAWEB06CgghEB4QDxAWEB06BwghEAoQoAFKBAhBGABKBAhGGABQ2gpYxmFglm1oBHABeACAAa4BiAHoHZIBBTE1LjIwmAEAoAEByAEJwAEB&sclient=gws-wiz

A good article:
https://betterprogramming.pub/the-use-of-the-javascript-semicolon-843fd5e94d0e
my principle is reliability. so I recommend - use. you can do it differently if you think it's better.